### ISSUE TITLE
Add Courier SDK to support multi-channel Communication

### ISSUE THREAD

**Issue by tk26** [hagopj13/node-express-boilerplate#167]
Disclaimer: I am an engineer at Courier, so this certainly has my own biases.

TL;DR: Add Courier SDK to support notifications out of the box via a single API invocation.

https://www.courier.com/ provides a single API to communicate across email, SMS, direct messaging and so on. This enables developers to configure their providers of choice (For ex. SendGrid for Email, Twilio for SMS, Discord for direct messaging etc) via a UI, design how they want their notification to look like, and invoke it via a single API call from their tech stack.

Enabling Courier API via this boilerplate will help developers tremendously. We believe you should never build 3 things in your app yourself - Authentication, Payments and Notifications [Ref: https://www.courier.com/blog/the-three-things-to-never-build-in-your-app/]
